mods/default/workbench.lua

```diff
@@ -0,0 +1,229 @@
+		"list[current_player;main;0,4.08;8,3;8]" ..
+		"field_close_on_enter[text;false]" ..
+		default.get_hotbar_bg(0,2.85)
+end
+
+local function get_item_desc(stack)
+	local desc = stack:get_meta():get_string("description")
+
+	if desc == "" then
+		desc = minetest.registered_items[stack:get_name()].description
+	end
+
+  return desc
+end
+
+local function workbench_update_text(pos, stack)
+  local meta = minetest.get_meta(pos)
+	meta:set_string("text", get_item_desc(stack))
+	meta:set_string("formspec", get_workbench_formspec(pos))
+end
+
+local function workbench_update_help(pos, type, string)
+	local meta = minetest.get_meta(pos)
+	meta:set_string("formspec", get_workbench_formspec(pos, string))
+	meta:set_string("error", type)
+end
+
+--
+-- Node definition
+--
+
+minetest.register_node("default:workbench", {
+	description = "Workbench",
+	tiles = {"default_workbench_top.png", "default_wood.png", "default_workbench_sides.png",
+		"default_workbench_sides.png", "default_workbench_sides.png", "default_workbench_sides.png"},
+	groups = {choppy = 3, oddly_breakable_by_hand = 2, flammable = 3},
+	sounds = default.node_sound_wood_defaults(),
+
+	drawtype = "nodebox",
+	paramtype = "light",
+	node_box = {
+		type = "fixed",
+		fixed = {
+			{-0.4375, 0.1875, -0.4375, 0.4375, 0.5, 0.4375}, -- Top
+			{-0.5, 0.25, -0.5, -0.4375, 0.5, 0.5}, -- TopSide1
+			{0.4375, 0.25, -0.5, 0.5, 0.5, 0.5}, -- TopSide2
+			{-0.5, 0.25, 0.4375, 0.5, 0.5, 0.5}, -- TopSide3
+			{-0.5, 0.25, -0.5, 0.5, 0.5, -0.4375}, -- TopSide4
+			{-0.4375, -0.5, 0.25, -0.25, 0.5, 0.4375}, -- Leg1
+			{-0.4375, -0.5, -0.4375, -0.25, 0.5, -0.25}, -- Leg2
+			{0.25, -0.5, 0.25, 0.4375, 0.5, 0.4375}, -- Leg3
+			{0.25, -0.5, -0.4375, 0.4375, 0.5, -0.25}, -- Leg4
+		}
+	},
+
+	on_construct = function(pos)
+		local meta = minetest.get_meta(pos)
+		meta:set_string("formspec", get_workbench_formspec(pos))
+		local inv = meta:get_inventory()
+		inv:set_size("input", 1)
+		inv:set_size("nametag", 1)
+		inv:set_size("output", 1)
+	end,
+	can_dig = function(pos, player)
+		local inv = minetest.get_meta(pos):get_inventory()
+
+		if inv:is_empty("input") and inv:is_empty("nametag") and inv:is_empty("output") then
+			return true
+		else
+			return false
+		end
+	end,
+	on_blast = function(pos)
+		local inv = minetest.get_meta(pos):get_inventory()
+
+		local drops = {
+			inv:get_list("input")[1],
+			inv:get_list("nametag")[1],
+			inv:get_list("output")[1],
+			"default:workbench",
+		}
+		minetest.remove_node(pos)
+		return drops
+	end,
+
+	allow_metadata_inventory_put = function(pos, listname, index, stack)
+		local inv = minetest.get_meta(pos):get_inventory()
+
+		if listname == "nametag" then
+			if stack:get_name() ~= "default:nametag" then
+				return 0
+			else
+				return stack:get_count()
+			end
+		elseif listname == "output" then
+			return 0
+		elseif listname == "input" then
+			if minetest.registered_items[stack:get_name()].groups.renameable == 0 then
+        return 0
```

Comment:
Your editor settings wrecked up somehow. Please look through the code and correct all of them (quite many).